IFTB.bs

```diff
@@ -0,0 +1,535 @@
+tables in the same order.
+
+Because the OpenType and WOFF2 formats are documented elsewhere, the initial
+IFTB font format is specified in terms of changes to and restrictions on those
+formats, including some specific OpenType tables.
+
+### Introduction ### {#font-organization-introduction}
+
+### Glyph Content ### {#iftb-glyph-content}
+
+The term <dfn>outline table</dfn> is used to describe these four tables, which carry different types of
+glyph outlines:
+
+-   The <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/cff"><code>CFF</code></a>
+    table
+-   The <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/cff2"><code>CFF2</code></a>
+    table
+-   The <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/glyf"><code>glyf</code></a>
+    table and its associated
+    <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/loca"><code>loca</code></a>table
+-   The <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/cff2"><code>gvar</code></a>
+    table
+
+An IFTB font must have either one CFF table, one CFF2 table, or one pair of glyf/loca tables.  A
+variable glyf-table-based font will also have a gvar table.
+
+### CFF and CFF2 Requirements and Recommendations ### {#iftb-cff-requirements}
+
+When an IFTB-encoded font includes a CFF or CFF2 table:
+
+1.   The CharStrings Index MUST be the last element of the table (followed by 0 padding to a four-byte
+     word boundary).
+2.   The CharStrings Index offset size MUST be 4.
+3.   The offset of the CharStrings index from the start of the CFF table MUST be stored accurately in
+     the relevant field of the IFTB table header.
+4.   All GIDs in the binned set must be represented in the index from the start, with trailing
+     zero-offset-change entries if needed.
+
+An "empty" glyph in a CFF table MUST be encoded as a single "endchar" operator. An empty glyph in a
+CFF2 table MUST be encoded as a zero-length string.
+
+CFF or CFF2 subroutinization is compatible with the IFTB format. However, <em>naive</em>
+subroutinization will tend to reduce chunk size while increasing the size of the initial file. Also,
+WOFF2 has often been shown to compress a non-subroutinized font more effectively than its subroutinized
+equivalent (at the cost of a significantly larger uncompressed file).  Therefore, it is recommended
+that subroutinzation either be avoided, used moderately, or customized to the specific needs of IFTB.
+
+### glyf/loca Requirements and Recommendations ### {#iftb-glyf-requirements}
+
+When an IFTB-encoded font includes glyf and loca tables:
+
+1.  The offset type of the loca table, specified in a field in the head table, MUST be the "Long"
+     version.
+2.  All GIDs in the binned set must be represented in the loca table from the start, with trailing
+     zero-offset-change entries if needed.
+3.  The tables must not be internally compressed.
```

Comment:
Is this referring to the optional woff2 transformation of loca and glyf? If so recommend specifically calling that out.